### PR TITLE
[hotfix] fix initialize bug with zero

### PR DESCRIPTION
**Pull request by feifeibear**

### CODE REVIEW

File: tests/test_zero_data_parallel/common.py [hunk 1]
```diff
@@ -8,7 +8,22 @@
...
+_ZERO_OFFLOAD_OPTIMIZER_CONFIG = dict(device='cpu', pin_memory=True, buffer_count=5, fast_init=False)
+_ZERO_OFFLOAD_PARAM_CONFIG = dict(device='cpu', pin_memory=True, buffer_count=5, buffer_size=1e8, max_in_cpu=1e9)
+
+ZERO_PARALLEL_CONFIG = dict(
+    fp16=dict(mode=None,),
+    # zero=dict(
+    #     optimizer=_ZERO_OPTIMIZER_CONFIG,
+    #     offload_optimizer_config=_ZERO_OFFLOAD_OPTIMIZER_CONFIG,
+    #     offload_param_config=_ZERO_OFFLOAD_PARAM_CONFIG,
+    # ),
```
Comment (FrankLeeeee) [inline tests/test_zero_data_parallel/common.py:23]: remove this?

File: tests/test_zero_data_parallel/test_zero_engine.py [hunk 2]
```diff
@@ -67,15 +76,30 @@ def run_dist(rank, world_size, port):
             torch_optimizer.step()
             i += 1
 
-        check_sharded_params_padding(torch_model, zero_model, loose=True)
+        # for torch_param, zero_param in zip(torch_model.parameters(), colo_model.parameters()):
+        #     assert torch.allclose(torch_param, zero_param), f"diff {torch_param - zero_param}"
```
Comment (FrankLeeeee) [inline tests/test_zero_data_parallel/test_zero_engine.py:80]: remove this?

File: tests/test_zero_data_parallel/common.py (hunk 1 above)
Comment (feifeibear) [reply tests/test_zero_data_parallel/common.py:23]: DONE

File: tests/test_zero_data_parallel/test_zero_engine.py (hunk 2 above)
Comment (feifeibear) [reply tests/test_zero_data_parallel/test_zero_engine.py:80]: DONE

File: tests/test_zero_data_parallel/common.py [hunk 3]
```diff
@@ -8,7 +8,22 @@
 from colossalai.utils import checkpoint
 from colossalai.zero.sharded_model import ShardedModelV2
 
-LOGGER = get_dist_logger()
+LOGGER = get_dist_logger('zero_test')
+
+_ZERO_OPTIMIZER_CONFIG = dict(optimizer_type=torch.optim.Adam, optimizer_config=dict(lr=1e-3))
+_ZERO_OFFLOAD_OPTIMIZER_CONFIG = dict(device='cpu', pin_memory=True, buffer_count=5, fast_init=False)
+_ZERO_OFFLOAD_PARAM_CONFIG = dict(device='cpu', pin_memory=True, buffer_count=5, buffer_size=1e8, max_in_cpu=1e9)
```
Comment (ver217) [inline tests/test_zero_data_parallel/common.py:15]: I think this is useless now.

Comment (feifeibear) [reply tests/test_zero_data_parallel/common.py:15]: DONE

File: colossalai/initialize.py [hunk 4]
```diff
@@ -268,17 +268,18 @@ def initialize(model: Union[Callable, nn.Module],
...
     use_zero = hasattr(gpc.config, 'zero')
     if use_zero:
         zero_cfg = gpc.config.get('zero', None)
         if zero_cfg is not None:
             cfg_ = zero_cfg.copy()
         else:
             cfg_ = {}
-        optimizer_config = zero_cfg.get('optimzer', None)
+        optimizer_config = zero_cfg.get('optimizer', None)
         model, optimizer = convert_to_zero_v2(model_builder=model, optimizer_config=optimizer_config)
```
Comment (ver217) [inline colossalai/initialize.py:280]: Offload config isn't passed here.

Comment (feifeibear) [reply colossalai/initialize.py]: DONE, redesign the config

**Review by ver217 — APPROVED** [review]
LGTM

**Pull request merged**